crates/swc_ecma_minifier/src/compress/optimize/bools.rs

```diff
@@ -167,34 +184,25 @@ impl Optimizer<'_> {
 
         let lt = left.get_type(self.ctx.expr_ctx);
         let rt = right.get_type(self.ctx.expr_ctx);
-        if let Value::Known(lt) = lt {
-            if let Value::Known(rt) = rt {
-                match (lt, rt) {
-                    (Type::Undefined, Type::Null) | (Type::Null, Type::Undefined) => {
-                        if op == op!("===") {
-                            report_change!(
-                                "Reducing `!== null || !== undefined` check to `!= null`"
-                            );
-                            return Some(BinExpr {
-                                span,
-                                op: op!("=="),
-                                left: cmp.take(),
-                                right: Lit::Null(Null { span: DUMMY_SP }).into(),
-                            });
-                        } else {
-                            report_change!(
-                                "Reducing `=== null || === undefined` check to `== null`"
-                            );
-                            return Some(BinExpr {
-                                span,
-                                op: op!("!="),
-                                left: cmp.take(),
-                                right: Lit::Null(Null { span: DUMMY_SP }).into(),
-                            });
-                        }
-                    }
-                    _ => {}
-                }
+        if let (Value::Known(Type::Undefined), Value::Known(Type::Null))
+        | (Value::Known(Type::Null), Value::Known(Type::Undefined)) = (lt, rt)
```

Comment:
```suggestion
        | (Value::Known(Type::Null | Type::Undefined)) = (lt, rt)
```

Reply:
This is to ensure they're `undefined | null` or `null | undefined`, but never `undefined | undefined` or `null | null`.

Reply:
Of course there would not be any real code like this.

Reply:
Ah. Thanks. I misread it.